### PR TITLE
Feature: error thrown when query not string

### PR DESCRIPTION
**Pull request by lewisPrescott707**
Feature: error when query string values are not strings

- Add check to interaction for query equals a string
- Add Typescript support for query as string[]

Tests:

- Added unit tests to check for all query types (were only being caught in example tests before)
- Moved karma examples to use Chrome-Headless rather than out-of-date phantomjs

Issue:
[#428](https://github.com/pact-foundation/pact-js/issues/428)

### CODE REVIEW

**Comment by mefellows** [comment]
Thanks for this Lewis! Unfortunately it's going to be a little more complicated, as query strings may be provided in a number of ways. They can be either a string, or an object where keys are strings and values are Matchers e.g.

```
      const interaction = {
        state: "i have a list of cats",
        uponReceiving: "a request for cats with given catId",
        withRequest: {
          method: "GET",
          path: "/cats",
          query: {
            "catId[]": Matchers.eachLike("1"),
          },
          headers: {
            Accept: "application/json",
          },
        },
        willRespondWith: {
          status: 200,
          headers: {
            "Content-Type": "application/json",
          },
          body: EXPECTED_BODY,
        },
      }
```

Would you be keen to tackle that scenario also?

**Comment by lewisPrescott707** [comment]
Yes I can tackle that.

**Comment by lewisPrescott707** [comment]
There appears to be a error happening across all PR's.

**Comment by mefellows** [comment]
@lewisPrescott707 apologies, yes, there appears to be an issue with one of the `pact-message` examples on Windows (there may be an issue in (via `pact-node`). Please disregard the appveyor builds for now, unless there is a specific aspect of the build relevant to you. It looks like the standard build (aka the framework itself) is ok.

Let me know when this is ready for review and I'll jump on it.

**Comment by lewisPrescott707** [comment]
Think it's ready for review now.

File: src/dsl/interaction.spec.ts [hunk 1]
```diff
@@ -124,6 +135,52 @@ describe("Interaction", () => {
...
+
+      it("is passed with array", () => {
+        request.query = {
+          id: ["1", "2"],
+        }
+        expect(
+          new Interaction()
+            .uponReceiving("request")
+            .withRequest(request)
+            .json().request?.query
```
Comment (mefellows) [inline src/dsl/interaction.spec.ts:179]: Did not realise the `?.` syntax was available - nice!

File: examples/karma/package.json [hunk 2]
```diff
@@ -17,9 +17,13 @@
...
+    "karma-chrome-launcher": "^3.1.0",
     "karma-jasmine": "^1.1.0",
     "karma-mocha": "~1.3.0",
-    "karma-phantomjs-launcher": "^1.0.4",
-    "mocha": "^6.1.4"
+    "mocha": "^6.1.4",
+    "puppeteer": "^1.18.1"
+  },
+  "dependencies": {
+    "nstall": "^0.2.0"
```
[the remark this answers is not in the document]
Comment (lewisPrescott707) [reply examples/karma/package.json:27]: My bad. Must have been a typo when installing

**Comment by lewisPrescott707** [comment]
@mefellows made the changes you suggested. Thanks! Should be good to go

**Comment by mefellows** [comment]
Awesome, thanks Lewis! I'll get this out in the next release - appreciate your support.

**Pull request merged**